### PR TITLE
feat: added clip_ frontend function

### LINKED ISSUE #23296: clip_

### PR DESCRIPTION
**Pull request by Aaryan562**
Close #23296

### CODE REVIEW

**Comment by Aaryan562** [comment]
@tomatillos pls review it. Thanks

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_math.py [hunk 1]
```diff
@@ -31,6 +64,41 @@ def test_paddle_ceil_(
...
+    *,
+    input_and_ranges,
+    frontend,
+    fn_tree,
+    test_flags,
+    backend_fw,
+    on_device,
+):
+    input_dtype, x, min_val, max_val = input_and_ranges
+    if min_val > max_val:
```
Comment (tomatillos) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_math.py:82]: You could put this logic in the helper function, might be a bit neater. Also as I understand, you are currently always only testing with max and min present, however we want to sometimes pass max, min, or both. You could check the tests for the torch implementation as they have similar logic.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_math.py [hunk 2]
```diff
@@ -1,6 +1,39 @@
...
+    x_dtype, x = draw(
+        helpers.dtype_and_values(
+            available_dtypes=helpers.get_dtypes("valid"),
+            shape=shape,
+            min_value=0,
+            max_value=50,
+        )
+    )
+    min = draw(
+        helpers.array_values(dtype=x_dtype[0], shape=(1,), min_value=0, max_value=25)
```
Comment (tomatillos) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_math.py:27]: Also can we test on a slightly bigger shape, no reason not to

Comment (Aaryan562) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_math.py]: I have done the changes by seeing how others have implemented the same function in other frontends. All the local test cases are passing successfully.

**Comment by Aaryan562** [comment]
@tomatillos

**Comment by tomatillos** [comment]
Looks good, thanks

**Pull request merged**